File: GeeksforGeeks/C++/Medium/Is Binary Number Multiple of 3/is-binary-number-multiple-of-3.cpp
//{ Driver Code Starts

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function template for C++
class Solution{
public:	
		
	int isDivisible(string s){
	    //complete the function here
        int state = 0;
        int transitions[2][3] = {
            {0, 2, 1}, 
            {1, 0, 2}
        };

        for (char c : s)
            state = transitions[c - '0'][state];
        
        return state == 0;
	}
};

//{ Driver Code Starts.
int main(){
    
    int t;
    cin >> t;
    while(t--){
        string s;
        cin >> s;
        Solution ob;
        cout << ob.isDivisible(s) << endl;
    }
return 0;
}


// } Driver Code Ends

File: GeeksforGeeks/Python3/Medium/Is Binary Number Multiple of 3/is-binary-number-multiple-of-3.py
#User function Template for python3
class Solution:
	def isDivisible(self, s):
		# code here
		return 1 if int(s,2)%3==0 else 0

#{ 
 # Driver Code Starts

#Initial Template for Python 3

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		s = input()
		ob = Solution()
		ans = ob.isDivisible(s)
		print(ans)

# } Driver Code Ends